pynucastro/networks/tests/test_python_custom_rate.py

```diff
@@ -0,0 +1,92 @@
+# unit tests for rates
+import pytest
+from pytest import approx
+
+import pynucastro as pyna
+
+
+class MyRate(pyna.Rate):
+    def __init__(self, reactants=None, products=None,
+                 r0=1.0, T0=1.0, nu=0):
+
+        # we set the chapter to custom so the network knows how to deal with it
+        self.chapter = "custom"
+
+        # call the Rate init to do the remaining initialization
+        super().__init__(reactants=reactants, products=products)
+
+        self.r0 = r0
+        self.T0 = T0
+        self.nu = nu
+
+    def function_string_py(self):
+        """return a string containing a python function that computes
+        the rate"""
+        fstring = ""
+        fstring += "@numba.njit()\n"
+        fstring += f"def {self.fname}(rate_eval, tf):\n"
+        fstring += f"    rate_eval.{self.fname} = {self.r0} * (tfactors.T9 * 1.e9 / {self.T0} )**({self.nu})\n\n"
```

Comment:
This doesn't look right...
```suggestion
        fstring += f"    rate_eval.{self.fname} = {self.r0} * (tf.T9 * 1.e9 / {self.T0} )**({self.nu})\n\n"
```

Reply:
indeed.  I missed that on the previous PR review

Reply:
We should probably write out the network and try evaluating the rhs

Reply:
indeed.  I'll add that

Reply:
okay, this now does an import and checks the ydots